pkg/apis/apps/validation/validation.go

```diff
@@ -149,7 +149,7 @@ func ValidateStatefulSetUpdate(statefulSet, oldStatefulSet *apps.StatefulSet) fi
 	restoreStrategy := statefulSet.Spec.UpdateStrategy
 	statefulSet.Spec.UpdateStrategy = oldStatefulSet.Spec.UpdateStrategy
 
-	if !reflect.DeepEqual(statefulSet.Spec, oldStatefulSet.Spec) {
+	if !apiequality.Semantic.DeepEqual(statefulSet.Spec, oldStatefulSet.Spec) {
```

Comment:
this is used at one more place:

https://github.com/kubernetes/kubernetes/blob/0fcc38425843d973538ce743a907e4c98095f8a0/pkg/apis/policy/validation/validation.go#L53-L55

Reply:
another sweep could be good, but I didn't see anything under that object that semantic equality would affect